Source/WebCore/css/parser/CSSSelectorParser.cpp

```diff
@@ -162,17 +157,22 @@ CSSSelectorList CSSSelectorParser::consumeForgivingComplexSelectorList(CSSParser
     });
 }
 
-bool CSSSelectorParser::supportsComplexSelector(CSSParserTokenRange range, const CSSParserContext& context)
+bool CSSSelectorParser::supportsComplexSelector(CSSParserTokenRange range, const CSSParserContext& context, IsNestedContext isNestedContext)
 {
     range.consumeWhitespace();
-    CSSSelectorParser parser(context, nullptr);
+    CSSSelectorParser parser(context, nullptr, isNestedContext);
 
     // @supports requires that all arguments parse.
     parser.m_disableForgivingParsing = true;
 
     // FIXME: https://bugs.webkit.org/show_bug.cgi?id=215635
     // Unknown css selector combinator is not addressed correctly in |CSSSelectorParser::consumeComplexSelector|.
-    auto parserSelector = parser.consumeComplexSelector(range);
+    std::unique_ptr<CSSParserSelector> parserSelector;
+    if (isNestedContext == IsNestedContext::Yes)
+        parserSelector = parser.consumeNestedComplexSelector(range);
+    else
+        parserSelector = parser.consumeComplexSelector(range);
```

Comment:
Channeling @darinadler :

```
auto parserSelector = isNestedContext == IsNestedContext::Yes ?  parser.consumeNestedComplexSelector(range) : parser.consumeComplexSelector(range);
```

Reply:
It does seem like `consumerNestedComplexSelector` always tries for `consumerComplexSelector` first, then tries the Relative case is that fails.

Maybe this would work:

```
auto parserSelector = parser.consumeComplexSelector(range);
if (!parserSelector && isNestedContext == IsNestedContext::Yes)
    parserSelector = consumeRelativeNestedSelector(range);
```

Reply:
It does, but I don't think we get any benefit in writing it that way (it requires more knowledge of what is a nested selector for the caller for no reason).